### PR TITLE
Only add explicit bool if we have C++11

### PR DESCRIPTION
**Pull request by JonathanMeans**
GCC 4.4.7 gave a compile error here.

### CODE REVIEW

File: modules/c++/mem/include/mem/ScopedCopyablePtr.h [hunk 1]
```diff
@@ -106,11 +106,18 @@ class ScopedCopyablePtr
     {
         return !(*this == rhs);
     }
-    
-    explicit operator bool() const 
-    { 
-        return get() == NULL ? false : true; 
+
+    // explicit operators not suppored until C++11
```
Comment (asylvest) [inline modules/c++/mem/include/mem/ScopedCopyablePtr.h]: suppored --> supported

File: modules/c++/mem/include/mem/ScopedCopyablePtr.h [hunk 2]
```diff
@@ -106,11 +106,18 @@ class ScopedCopyablePtr
...
+    bool toBool() const
+    {
+        return get() == NULL ? false : true;
     }
+#ifdef __CODA_CPP11
```
Comment (asylvest) [inline modules/c++/mem/include/mem/ScopedCopyablePtr.h]: You need to add one of the _config headers to get this

Comment (gpotts) [reply modules/c++/mem/include/mem/ScopedCopyablePtr.h]: For non CPP11 this should be valid

#ifdef __CODA_CPP11
    explicit operator bool() const
    {
        return toBool();
    }
#else
    operator bool() const
    {
        return toBool();
    }
#endif

Comment (asylvest) [reply modules/c++/mem/include/mem/ScopedCopyablePtr.h]: OK so in that case @JonathanMeans can you put out a new PR that collapses this to...
<remove `toBool()`>

```
#ifdef _CODA_CPP11
explicit
#endif
operator bool() const
{
    return get() == NULL ? false : true;
}
```

**Pull request merged**